### PR TITLE
chore: Upgrade argocd-operator versions

### PR DESCRIPTION
**Pull request by ishitasequeira**
**What type of PR is this?**
/kind chore

**What does this PR do / why we need it**:
Upgrade argocd-operator versions

**Test acceptance criteria**:

* [x] Unit Test
* [x] E2E Test

### CODE REVIEW

**Comment by svghadi** [comment]
Can you also update the CRDs here?
https://github.com/redhat-developer/gitops-operator/tree/master/config/crd/bases

File: test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ commands:
       # These variables need to be maintained according to the component matrix: https://spaces.redhat.com/display/GITOPS/GitOps+Component+Matrix
       expected_kustomizeVersion='v5.4.2'
       expected_helmVersion='v3.15.2'
-      expected_argocdVersion='v2.12.6+4dab5bd'
+      expected_argocdVersion='v2.12.7'
```
Comment (anandf) [inline test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml:11]: There should be a short commit id too added isn't it ?

Comment (ishitasequeira) [reply test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml:11]: This needs to be matched to the version in `go.mod` which is automatically updated when running `go mod download && go mod tidy && go mod vendor`

**Comment by svghadi** [comment]
> @svghadi , the CRDs seem to be already updated. That is why no change came in the CRDs.

@ishitasequeira - Can you confirm this again? I believe there is a CRD change. In your https://github.com/argoproj-labs/argocd-operator/pull/1591 PR, I see a new `autoHealAttemptsCount` field in Application CRD which is not present in `v1.14` branch in this repo.

File: test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml (hunk 1 above)
Comment (anandf) [reply test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml:11]: I think the test will fail as it will try to match `v2.12.7`

Comment (ishitasequeira) [reply test/openshift/e2e/parallel/1-031_validate_toolchain/01-check.yaml:11]: The test seems to be succeeding. The version in `go.mod` is also reflecting `v2.12.7` matching the value in test.

**Review by anandf — APPROVED** [review]
LGTM

**Review by anandf — APPROVED** [review]
LGTM

**Pull request merged**